traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -699,62 +759,301 @@ WHERE
 }
 
 func updateV15(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, reqDS *tc.DeliveryServiceNullableV15) (*tc.DeliveryServiceNullableV15, int, error, error) {
-	dsV30 := tc.DeliveryServiceNullableV30{DeliveryServiceNullableV15: *reqDS}
+	dsV31 := tc.DeliveryServiceV31{
+		DeliveryServiceV30: tc.DeliveryServiceV30{
+			DeliveryServiceNullableV15: *reqDS,
+		},
+	}
 	// query the DB for existing 3.0 fields in order to "upgrade" this 1.5 request into a 3.0 request
 	query := `
 SELECT
   ds.topology,
   ds.first_header_rewrite,
   ds.inner_header_rewrite,
   ds.last_header_rewrite,
-  ds.service_category
+  ds.service_category,
+  ds.max_request_header_bytes
 FROM
   deliveryservice ds
 WHERE
   ds.id = $1`
 	if err := inf.Tx.Tx.QueryRow(query, *reqDS.ID).Scan(
-		&dsV30.Topology,
-		&dsV30.FirstHeaderRewrite,
-		&dsV30.InnerHeaderRewrite,
-		&dsV30.LastHeaderRewrite,
-		&dsV30.ServiceCategory,
+		&dsV31.Topology,
+		&dsV31.FirstHeaderRewrite,
+		&dsV31.InnerHeaderRewrite,
+		&dsV31.LastHeaderRewrite,
+		&dsV31.ServiceCategory,
+		&dsV31.MaxRequestHeaderBytes,
 	); err != nil {
 		if err == sql.ErrNoRows {
-			return nil, http.StatusNotFound, fmt.Errorf("delivery service ID %d not found", *dsV30.ID), nil
+			return nil, http.StatusNotFound, fmt.Errorf("delivery service ID %d not found", *dsV31.ID), nil
 		}
-		return nil, http.StatusInternalServerError, nil, fmt.Errorf("querying delivery service ID %d: %s", *dsV30.ID, err.Error())
+		return nil, http.StatusInternalServerError, nil, fmt.Errorf("querying delivery service ID %d: %s", *dsV31.ID, err.Error())
 	}
-	res, status, userErr, sysErr := updateV30(w, r, inf, &dsV30)
+	res, status, userErr, sysErr := updateV31(w, r, inf, &dsV31)
 	if res != nil {
 		return &res.DeliveryServiceNullableV15, status, userErr, sysErr
 	}
 	return nil, status, userErr, sysErr
 }
 
-func updateV30(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, ds *tc.DeliveryServiceNullableV30) (*tc.DeliveryServiceNullableV30, int, error, error) {
+func updateV31(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, ds *tc.DeliveryServiceV31) (*tc.DeliveryServiceV31, int, error, error) {
```

Comment:
similar to my comment for `createV31`, we should rename this parameter to `dsV31`, then use `ds` instead of `dsNullableV30` on L803